lib/model/autocomplete.dart

```diff
@@ -244,11 +246,102 @@ class MentionAutocompleteView extends ChangeNotifier {
     notifyListeners();
   }
 
+  List<User>? _sortedUsers;
+
+  int compareByDms(User userA, User userB) {
+    final aRecencyIndex = store.recentDmConversationsView.getRecencyIndex(userA.userId);
+    final bRecencyIndex = store.recentDmConversationsView.getRecencyIndex(userB.userId);
+
+    if (aRecencyIndex > bRecencyIndex) {
+      return -1;
+    } else if (bRecencyIndex > aRecencyIndex) {
+      return 1;
+    }
+
+    if (!userA.isBot && userB.isBot) {
+      return -1;
+    } else if (userA.isBot && !userB.isBot) {
+      return 1;
+    }
```

Comment:
This part doesn't seem like it belongs in a method named `compareByDms`, as Greg pointed out above: https://github.com/zulip/zulip-flutter/pull/608#discussion_r1554283946